golem-worker-executor-base/src/services/worker.rs

```diff
@@ -236,8 +242,26 @@ impl WorkerService for DefaultWorkerService {
                         panic!("failed to get worker status for {owned_worker_id} from KV storage: {err}")
                     });
 
-                if let Some(status) = status_value {
-                    details.last_known_status = status;
+                match status_value {
+                    Some(Ok(status)) => {
+                        details.last_known_status = status;
+                    }
+                    // We had a status, but it was written in a previous format and is not longer valid -> recompute
+                    Some(Err(_)) => {
```

Comment:
This error here is only for deserializtion -> we are dealing with an old version.